libtock/udp.c

```diff
@@ -90,12 +90,12 @@ ssize_t udp_send_to(void *buf, size_t len,
   if (!sval.success) return tock_status_to_returncode(sval.status);
 
   syscall_return_t ret = command(UDP_DRIVER, COMMAND_SEND, 0, 0);
-  if (ret.type == TOCK_SYSCALL_SUCCESS) {
-    return RETURNCODE_SUCCESS;
+  if (ret.type == TOCK_SYSCALL_FAILURE) {
+    return tock_status_to_returncode(ret.data[0]);
```

Comment:
Is this right? Before the code would only `yield_for()` if the command failed. Now the code only `yield_for()` if the command succeeds. This new version does seem like normal behavior, I just wanted to check.

Reply:
Yeah the old version did not work, which I caught in release testing. It looks like when porting to 2.0 I made a mistake and changed the logic. I didn't just sneak that change into this PR, it was part of #235 which this depended on.

Reply:
Cool. I was just checking for things that seem weird. As long as you've thought it through and tested it, then it's good.